habitat/datasets/object_nav/object_nav_dataset.py

```diff
@@ -30,7 +30,23 @@ class ObjectNavDatasetV1(PointNavDatasetV1):
     content_scenes_path: str = "{data_path}/content/{scene}.json.gz"
 
     def to_json(self) -> str:
+        self.goals_per_category = {}
```

Comment:
If we will have this code in `from_json`, then `self.goals_per_category` will be save here automatically.